### PR TITLE
On windows there can be problems with Tools that do not support Unicode

### PR DESCRIPTION
**Pull request by hknielsen**
Having a Username containing unicode chars, and tools that does not handle that well, makes it difficult when ToolTask are defined from NuGet Packages.

Fixes https://github.com/dotnet/msbuild/issues/9229

### Context
Tmp folder can be changed by updating the User env vars, but Username are not possible. Not prepending the Username makes it possible for Users to handle these rare cases.

### Changes Made
Changed to not append Username on Windows to tmp rsp file location.

### Testing
Tests already for ToolTasks using this.

### CODE REVIEW

**Comment by hknielsen** [comment]
@KalleOlaviNiemitalo is it Runtime check for windows are used on MsBuild or do we have preprocessors for it?

**Comment by hknielsen** [comment]
@KalleOlaviNiemitalo pushed new change, thanks!
Now we just need to determine if we should use the `SecurityIdentifier` - not sure how important it is

**Comment by rainersigwald** [comment]
I think this is a reasonable approach, and we'll check with security folks. Another option would be "on Windows, coerce the username to ANSI".

That said, I think there's room for https://github.com/dotnet/msbuild/issues/9229 to be fixed in `ToolTask` itself too, since the "change TEMP then call task" dance is pretty awkward.

**Comment by hknielsen** [comment]
> I think this is a reasonable approach, and we'll check with security folks. Another option would be "on Windows, coerce the username to ANSI".
Sound good, let me know. we can do the conversion to ANSI or use `SecurityIdentifier` depending on that.

> That said, I think there's room for #9229 to be fixed in `ToolTask` itself too, since the "change TEMP then call task" dance is pretty awkward.
I agree :) Lets discuss possible solutions on that Issue

File: src/Shared/TempFileUtilities.cs [hunk 1]
```diff
@@ -36,7 +38,18 @@ internal static void ClearTempFileDirectory()
...
+            string msbuildTempFolder;
+            // On windows Username with Unicode chars can give issues, so we dont append username to the temp folder name.
+            if (NativeMethodsShared.IsWindows)
+            {
+                msbuildTempFolder = msbuildTempFolderPrefix;
+            }
+            else
+            {
+                msbuildTempFolder = msbuildTempFolderPrefix + Environment.UserName;
+            }
```
Comment (Forgind) [inline src/Shared/TempFileUtilities.cs:50]: nit:
```suggestion
            // On windows Username with Unicode chars can give issues, so we dont append username to the temp folder name.
            string msbuildTempFolder = NativeMethodsShared.IsWindows ?
                msbuildTempFolderPrefix :
                msbuildTempFolderPrefix + Environment.UserName;
```

I'm a big fan of ternary operators, especially at initialization time like this, but I recognize not everyone agrees 🙂

**Review by Forgind — APPROVED** [review]
Looks good to me! Approval is pending security review as rainersigwald mentioned.

File: src/Shared/TempFileUtilities.cs (hunk 1 above)
Comment (hknielsen) [reply src/Shared/TempFileUtilities.cs:50]: Depends on the situation, but here it makes perfect sense :) thanks!

**Review by JanKrivanek — APPROVED** [review]
LGTM

**Review by AR-May — APPROVED** [review]

**Pull request merged**